sequencer/src/state.rs

```diff
@@ -754,6 +706,64 @@ impl ValidatedState {
 
         Ok((validated_state, delta))
     }
+
+    /// Updates the `ValidatedState` if a protocol upgrade has occurred.
+    pub(crate) fn apply_upgrade(
+        &mut self,
+        instance: &NodeState,
+        header: &Header,
+        version: Version,
+    ) {
+        // Check for protocol upgrade based on sequencer version
+        if version <= instance.current_version {
+            return;
+        }
+
+        let Some(upgrade) = instance.upgrades.get(&version) else {
+            return;
+        };
+
+        match upgrade.upgrade_type {
+            UpgradeType::ChainConfig { chain_config } => {
+                if header.chain_config.commit() == chain_config.commit() {
```

Comment:
Remove this check. We're applying the upgrade whether the proposed header has done it or not, because that's what the protocol says we should do. We will later fail validation if the header hasn't done it.

This should be analogous to how we handle, e.g., the block merkle tree: we push the new block onto the tree unconditionally, because that's the protocol. Then we later fail validation if the proposed header got a different blocks merkle root than we did, because it means the proposer didn't follow the protocol

Reply:
Yeah, I added this check back when there were a lot of view timeouts and I would see chain config mismatch errors in an infinite loop. Now, the test passes without it.

Reply:
removed in ef2e4913018cf3d088423e8b364d0e4f3ff87485

Reply:
The check is still here?